File: UniBlock/src/main/scala/persistence/KeyManager.scala
package it.unifi.nave.uniblock
package persistence

import java.security.PrivateKey

trait KeyManager {
  def saveDhPk(id: String, pk: PrivateKey): Unit

  def saveSignPk(id: String, pk: PrivateKey): Unit

  def retrieveDhPk(id: String): Option[PrivateKey]

  def retrieveSignPk(id: String): Option[PrivateKey]
}

File: program/src/main/java/it/unifi/nave/uniblock/persistence/KeyManager.java
package it.unifi.nave.uniblock.persistence;

import java.security.PrivateKey;

public interface KeyManager {
  void saveDhPk(String id, PrivateKey pk);

  void saveSignPk(String id, PrivateKey pk);

  PrivateKey retrieveDhPk(String id);

  PrivateKey retrieveSignPk(String id);
}
